sc-poc/managers/configman/common/http.go

```diff
@@ -87,59 +89,78 @@ func getAPIRoutes() caddyhttp.Route {
 	}
 }
 
-func getConfigRoutes() caddyhttp.Route {
+func getConfigRoutes(config ConfigType) caddyhttp.Route {
 	// Make route list for the sub router
-	routeList := caddyhttp.RouteList{
-		// Open API for the config and operation endpoints
-		caddyhttp.Route{
-			Group:          "openapi",
-			MatcherSetsRaw: utils.GetCaddyMatcherSet([]string{"/v1/config/openapi.json"}, []string{http.MethodGet}),
-			HandlersRaw:    utils.GetCaddyHandler("config_openapi", nil),
-		},
-
-		// Admin auth middleware
-		caddyhttp.Route{
-			Group:       "auth",
-			HandlersRaw: utils.GetCaddyHandler("admin_auth", nil),
-		},
-
-		// Config routes
-		caddyhttp.Route{
-			Group:          "config",
-			MatcherSetsRaw: utils.GetCaddyMatcherSet([]string{"/v1/config/*"}, []string{http.MethodGet}),
-			HandlersRaw:    utils.GetCaddyHandler("config_get", nil),
-		},
-		caddyhttp.Route{
-			Group:          "config",
-			MatcherSetsRaw: utils.GetCaddyMatcherSet([]string{"/v1/config/*"}, []string{http.MethodDelete}),
-			HandlersRaw:    utils.GetCaddyHandler("config_delete", nil),
-		},
-		caddyhttp.Route{
-			Group:          "config",
-			MatcherSetsRaw: utils.GetCaddyMatcherSet([]string{"/v1/config/*"}, []string{http.MethodPost}),
-			HandlersRaw:    utils.GetCaddyHandler("config_apply", nil),
-		},
-
-		// Operation routes
-		caddyhttp.Route{
-			Group:          "operation",
-			MatcherSetsRaw: utils.GetCaddyMatcherSet([]string{"/v1/operation/*"}, []string{}),
-			HandlersRaw:    utils.GetCaddyHandler("operation", nil),
-		},
+	configRoutes := caddyhttp.RouteList{}
+	for k, v := range config {
+		gvr := source.GetResourceGVR(k)
+
+		// Create route of GVR for List operation
+		gvrPath := createConfigGVRPath(gvr)
+		data := make(map[string]interface{})
+		data["gvr"] = gvr
+
+		// Route for List operation
+		listRoute := caddyhttp.Route{
+			Group:          "config_list",
+			MatcherSetsRaw: utils.GetCaddyMatcherSet([]string{gvrPath}, []string{http.MethodGet}),
+			HandlersRaw:    utils.GetCaddyHandler("config_list", data),
+		}
+
+		// Route for Apply operation
+		applyRoute := caddyhttp.Route{
+			Group:          "config_apply",
+			MatcherSetsRaw: utils.GetCaddyMatcherSet([]string{gvrPath}, []string{http.MethodPut}),
+			HandlersRaw:    utils.GetCaddyHandler("config_apply", data),
+		}
+
+		configRoutes = append(configRoutes, listRoute, applyRoute)
+
+		// Create route of each instances of GVR for Get, Apply and Delete operations
+		for _, unstr := range v {
```

Comment:
I suggest we don't loop here. It will create multiple routes

Reply:
I didn't understand. How can I create the `Get` and `Delete` route for each source?